sw/host/opentitanlib/src/transport/hyperdebug/gpio.rs

```diff
@@ -268,7 +442,29 @@ impl GpioMonitoring for HyperdebugGpioMonitoring {
         }
         Ok(MonitoringReadResponse {
             events,
-            timestamp: reference_time, // TODO: adjust in case of event later than this timestamp
+            timestamp: reference_time,
         })
     }
 }
+
+/// Read 7 bits from each byte, least significant byte first.  High bit of one indicates more
+/// bytes belong to the same value.
+fn decode_leb128(idx: &mut usize, databytes: &[u8]) -> Result<u64> {
+    let mut i = *idx;
+    let mut value = 0u64;
+    let mut shift = 0;
+    while i < databytes.len() {
+        let byte = databytes[i];
+        value |= ((byte & 0x7F) as u64) << shift;
```

Comment:
This may overflow there are 10 bytes without MSB set.

Reply:
I have inserted a check to return an error in case of a tenth byte in the encoding, meaning that the routine accepts at most 63 bits of data for an u64.  I do not think it is worth supporting the most significant 64th bit being set through the lowest bit in the tenth byte of encoding, as it would make error handling more complicated.